isthmus/src/main/java/io/substrait/isthmus/SubstraitRelVisitor.java

```diff
@@ -196,6 +197,19 @@ public Rel visit(LogicalJoin join) {
 
   @Override
   public Rel visit(LogicalCorrelate correlate) {
+    // left input of correlated-join is similar to the left input of a logical join
+    apply(correlate.getLeft());
+
+    // right input of correlated-join is similar to a correlated sub-query
+    apply(correlate.getRight());
+
+    var joinType =
+        switch (correlate.getJoinType()) {
+          case INNER -> Join.JoinType.INNER; // corresponds to CROSS APPLY join
+          case LEFT -> Join.JoinType.LEFT; // corresponds to OUTER APPLY join
+          default -> throw new UnsupportedOperationException(
```

Comment:
Could `JoinType` be `RIGHT` ?   Why do we through `UnsupportedOperationException` for other join type?

Reply:
`APPLY` is a nested-loop join, it evaluates the `RIGHT` expression for each record in the `LEFT` relation. Hence, I don't think other join-types, including `RIGHT`, are valid for `APPLY`.

Reply:
For reference please refer to [SQL dialect recognized by Calcite’s default SQL parser](https://calcite.apache.org/docs/reference.html). Below is text relevant to the PR. A `JOIN` can be `left or right` but `APPLY` can only be `OUTER` which maps to `LEFT`.
> tableExpression [ NATURAL ] [ { LEFT | RIGHT | FULL } [ OUTER ] ] JOIN tableExpression [ joinCondition ]
   tableExpression [ CROSS | OUTER ] APPLY tableExpression